packages/block/src/header.ts

```diff
@@ -558,13 +579,85 @@ export class BlockHeader {
         throw new Error('uncle block has a parent that is too old or too young')
       }
     }
+
+    if (this._common.isActivatedEIP(1559)) {
+      if (!this.baseFeePerGas) {
+        throw new Error('EIP1559 block has no base fee field')
+      }
+      let parentBaseFee = parentHeader.baseFeePerGas
+      const isInitialEIP1559Block = this._common.getEIPActivationBlockNumber(1559)!.eq(this.number)
```

Comment:
This is at the end a question of the hardfork and the respective block can be retrieved directly with the existing `common.hardforkBlock()` function and the context can still be given with the variable name, so nothing would be lost here by using the following I would say:

```typescript
const isInitialEIP1559Block = this._common.hardforkBlock('london').eq(this.number)
```

Not some super strong opinion here, but I guess there is no need for this additional `getEIPActivationBlockNumber()` function to be introduced and this can be removed from `Common`. Or co you have got a preference to keep for some reason? Is there any value to be able to retrieve this directly by EIP number?

Reply:
Update: ok, just seeing `Block` library `header.ts` that there is the following code where I cannot make too much out of it directly:

```typescript
// Small hack to hack in the activation block number
// (Otherwise there would be need for a custom chain only for testing purposes)
common.getEIPActivationBlockNumber = function (eip: number) {
  if (eip == 1559) {
    return new BN(1)
  }
}
```

Will leave this for now. Still think method should likely be removed though.

Reply:
I agree not adding an additional method to common if not needed, @jochem-brouwer what do you think about using @holgerd77's suggestion of `hardforkBlock`:

```typescript
const isInitialEIP1559Block = this._common.hardforkBlock('london').eq(this.number)
```

for testing purposes you can probably still stub it in the test suite itself.

Reply:
Missed the comment here, ok, I will the remove the added method from `Common`.